pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -254,6 +258,22 @@ func getNetworkIDsByFilter(is *InstanceService, opts *networks.ListOpts) ([]stri
 	return uuids, nil
 }
 
+func CreatePort(is *InstanceService, name string, net ProviderNetwork, securityGroups *[]string) (ports.Port, error) {
+	portCreateOpts := ports.CreateOpts{
+		Name:           name,
+		NetworkID:      net.networkID,
```

Comment:
Can we make `NetworkID` optional?

Reply:
Done :)

Reply:
Here is the the way things shake out:
Passing both a uuid and a subnet_id: Expect the user knows what they are doing, and let it fail if not. attempts to create port in the subnet in the network specified. openstack will get mad at them if they did something wrong here
Passing Filters and subnet_id: Redundant lookup, uses subnet id to lookup network id. Not sure why anyone would do this, but if they do it will work fine as long as subnet_id is valid.
subnet_id: uses subnet_id to lookup network id
Multiple Networks: exactly what you think it does